.github/workflows/test-with-allure-report.yml

```
@@ -30,21 +30,21 @@ jobs:
-      ANDROID_API_LEVEL: 28
+      ANDROID_API_LEVEL: 29
       ANDROID_ARCH: x86_64
 
     concurrency: test-android-${{ github.event.inputs.environment == '' && 'stage' || github.event.inputs.environment }}
     name: Run tests against Android
-    runs-on: macos-12
+    runs-on: macos-14
     outputs:
       maven_tests: ${{ steps.maven_tests.outcome }}
     steps:
       - uses: actions/checkout@v3
```

actionlint (1.7.10)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)